odoo/tools/js_sourcemap.py

```diff
@@ -0,0 +1,171 @@
+from functools import lru_cache
+
+
+class SourceMapGenerator:
+    """
+    The SourceMapGenerator creates the sourcemap maps the asset bundle to the js files.
+
+    What is a sourcemap ? (https://developer.mozilla.org/en-US/docs/Tools/Debugger/How_to/Use_a_source_map)
+    In brief: a source map is what makes possible to debug your processed/compiled/minified code as if you were
+    debugging the original, non-altered source code. It is a file that provides a mapping original <=> processed for
+    the browser to read.
+
+    This implementation of the SourceMapGenerator is a translation and adaptation of this implementation
+    in js https://github.com/mozilla/source-map. For performance purposes, we have removed all unnecessary
```

Comment:
ok, license (BSD) and copyright can be retrieved in the original project, I don't think we need to copy them here